### PR TITLE
Add specificHumidity to ADPUPA BUFR DUMP Python API

### PR DESCRIPTION
**Pull request by PraveenKumar-NOAA**
This PR adds specificHumidity to the ADPUPA BUFR DUMP Python API. The following file has been changed: ~/ush/ioda/bufr2ioda/bufr2ioda_adpupa.py

The specificHumidity is calculated using the dewPoint and pressure. The information is extracted from the prepBUFR codes and also documented. 

The validation of the output obs is performed using the specificHumidity from the prepBUFR data. For more details, please check: #1193

### CODE REVIEW

**Review by nicholasesposito — APPROVED** [review]
Looks good

File: ush/ioda/bufr2ioda/bufr2ioda_adpupa.py [hunk 1]
```diff
@@ -191,6 +206,7 @@ def bufr_to_ioda(config, logger):
     dewpointTemperatureOE = np.float32(np.ma.masked_array(np.full((len(tmdp)), 0.0)))
     windEastwardOE = np.float32(np.ma.masked_array(np.full((len(wspd)), 0.0)))
     windNorthwardOE = np.float32(np.ma.masked_array(np.full((len(wspd)), 0.0)))
+    specificHumidityOE = np.float32(np.ma.masked_array(np.full((len(qob)), 0.0)))
```
Comment (RussTreadon-NOAA) [inline ush/ioda/bufr2ioda/bufr2ioda_adpupa.py:209]: @PraveenKumar-NOAA , `bufr2ioda_adpupa.py` aborted on WCOSS2 (Cactus) on this line with the following traceback
```
  File "/lfs/h2/emc/da/noscrub/russ.treadon/git/global-workflow/ensda_obs/sorc/gdas.cd/ush/ioda/bufr2ioda/bufr2ioda_adpupa.py", line 209, in bufr_to_ioda
    specificHumidityOE = np.float32(np.ma.masked_array(np.full((len(qob)), 0.0)))
UnboundLocalError: local variable 'qob' referenced before assignment
```

This is a valid error.  `qob` is not assigned values until line 227
```
    qob = Compute_SpecificHumidity_from_dewPoint_and_Pressure(tmdp, prlc)
    logger.debug(f'   qob min/max = {qob.min()} {qob.max()}')
```

As a test I moved line 209 after line 227
```
    qob = Compute_SpecificHumidity_from_dewPoint_and_Pressure(tmdp, prlc)
    logger.debug(f'   qob min/max = {qob.min()} {qob.max()}')
    specificHumidityOE = np.float32(np.ma.masked_array(np.full((len(qob)), 0.0)))        
```
This worked on Cactus.

Is this an acceptable solution or is there a better solution?

Comment (PraveenKumar-NOAA) [reply ush/ioda/bufr2ioda/bufr2ioda_adpupa.py:209]: @RussTreadon-NOAA this is acceptable, thanks.

Comment (RussTreadon-NOAA) [reply ush/ioda/bufr2ioda/bufr2ioda_adpupa.py:209]: Thank you @PraveenKumar-NOAA for the confirmation.  I will add this fix to GDASApp PR #1252

**Pull request merged**